wire-grpc-client/api/wire-grpc-client.api

```diff
@@ -60,6 +60,14 @@ public final class com/squareup/wire/GrpcMethod {
 	public final fun getResponseAdapter ()Lcom/squareup/wire/ProtoAdapter;
 }
 
+public final class com/squareup/wire/GrpcResponse {
```

Comment:
The biggest change in this PR is to create this wrapper adapter around `okhttp3.Response`. The reasoning is that with Kotlin 1.9.20, an `actual` implementation can't provide default argument values within an `actual` method implementation when using typealiases to fulfill the implementation. The definition for `Response.header(…)` includes the parameter `defaultValue: String? = null`, and so we can no longer do `actual type alias GrpcResponse = okhttp3.Response`.

Instead we create a concrete `GrpcResponse` class within the JVM platform implementation which wraps `okhttp3.Response`.